ludwig/config_validation/checks.py

```diff
@@ -639,6 +640,31 @@ def check_qlora_requirements(config: "ModelConfig") -> None:  # noqa: F821
         raise ConfigValidationError("Fine-tuning and LLM with quantization requires using the 'lora' adapter")
 
 
+@register_config_check
+def check_qlora_merge_and_unload_compatibility(config: "ModelConfig") -> None:  # noqa: F821
+    """Checks that model.merge_and_unload() is supported by underlying model.save_pretrained() when merging QLoRA
+    layers."""
+    if config.model_type != MODEL_LLM or config.trainer.type == "none":
+        return
+
+    if not (
+        config.adapter
+        and config.adapter.type == "lora"
```

Comment:
lora or adalora?

Reply:
@arnavgarg1 Thank you for the great catch -- fixed.